Source/MediaInfo/Reader/Reader_libcurl.cpp

```diff
@@ -101,7 +101,12 @@ namespace Http
             CutHead (Host,  User,       "@"     );
             CutTail (Host,  Path,       "/"     );
             CutTail (User,  Password,   ":"     );
-            CutTail (Host,  Port,       ":"     );
+            if (Host.find("]")!=(size_t)-1) {
+                CutTail (Host,  Port,       "]:"    );
+                if (Host.find("]")==(size_t)-1) Host += "]";
+            }
+            else
+                CutTail (Host,  Port,       ":"     );

```

Comment:
```suggestion
            CutWithBracketsColumn(Host, Port);
```

with 
```c++
    static void CutWithBracketsColumn(std::string &Input, std::string &Output)
    {
        size_t BracketCount = 0;
        const auto Input_Size = Input.size();
        for (size_t i = 0; i < Input_Size; i++) {
            switch (Input[i])
            {
            case '[':
                BracketCount++;
                break;
            case ']':
                if (!BracketCount)
                    return;
                BracketCount--;
                break;
            case ':':
                if (!BracketCount) {
                    Output = Input.substr(i + 1);
                    Input.resize(i);
                    return;
                }
                break;
            }
        }
    }
```

After `CutTail`definition.
It seems more error proof e.g. with your code an extra `]` which does not exist at the end is added with some malformed strings, I prefer to avoid "fantom" characters and to put all in `Host` when it is not well understood.